rust/arrow/src/datatypes.rs

```diff
@@ -56,11 +56,44 @@ pub enum DataType {
     Float16,
     Float32,
     Float64,
+    Timestamp(TimestampUnit),
+    Date(DateUnit),
+    Time32(TimeUnit),
+    Time64(TimeUnit),
+    Interval(IntervalUnit),
     Utf8,
     List(Box<DataType>),
     Struct(Vec<Field>),
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub enum TimestampUnit {
+    Second,
+    Millisecond,
```

Comment:
Why we need a `TimestampUnit` given we already have `TimeUnit`? they are exactly the same.

Reply:
Good point. I checked the C++ code and they just have TimeUnit so I removed TimestampUnit now.